it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.datastream;
+
+import com.google.cloud.datastream.v1.*;
+import com.google.protobuf.Duration;
+import com.google.protobuf.FieldMask;
+import org.apache.arrow.util.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Default class for implementation of {@link DatastreamResourceManager} interface.
+ */
+public final class DefaultDatastreamResourceManager implements DatastreamResourceManager {
```

Comment:
Part of my last comment, this no longer needs to implement the DatastreamResourceManager interface, but should implement ResourceManager. It also does not need the "Default" prefix.
```suggestion
public final class DatastreamResourceManager implements ResourceManager {
```